src/util/simplify_expr.cpp

```diff
@@ -361,11 +359,38 @@ static simplify_exprt::resultt<> simplify_string_index_of(
 
     const array_exprt &s2_data = s2_data_opt->get();
 
-    it = std::search(
-      std::next(s1_data.operands().begin(), starting_index),
-      s1_data.operands().end(),
-      s2_data.operands().begin(),
-      s2_data.operands().end());
+    if(search_from_end)
+    {
+      auto starting_offset = 0;
+      if(starting_index > 0)
+        starting_offset = search_string_size - starting_index;
+
+      auto it = std::search(
+        std::next(s1_data.operands().rbegin(), starting_offset),
+        s1_data.operands().rend(),
+        s2_data.operands().rbegin(),
+        s2_data.operands().rend());
+
+      if(it != s1_data.operands().rend())
+      {
+        // Zero-indexed arrays, so -1.
+        return from_integer(
+          std::distance(s1_data.operands().begin(), it.base()) - 1,
+          expr.type());
+      }
```

Comment:
should return `-1` in the other case no? :tangerine:

Reply:
Would think so, but no, the other case works fine without it. Assume it's notched along +1 due to the way the reverse iterator works.